aten/src/ATen/native/native_functions.yaml

```diff
@@ -681,15 +681,29 @@
   structured_delegate: all.out
   variants: function, method
 
+- func: all.dims(Tensor self, int[1]? dim=None, bool keepdim=False) -> Tensor
+  device_check: NoCheck   # TensorIterator
+  structured_delegate: all.dims_out
+  variants: function, method
+  cpp_no_default_args: ['dim']
+  dispatch:
+    CompositeExplicitAutograd: all_dims_default
+
 - func: all.out(Tensor self, int dim, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
   device_check: NoCheck   # TensorIterator
   structured: True
-  precomputed:
-  - dim -> int dim
   dispatch:
     CPU, CUDA: all_out
     MPS: all_out_mps
 
+- func: all.dims_out(Tensor self, int[1]? dim=None, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
+  device_check: NoCheck   # TensorIterator
+  structured: True
+  dispatch:
+    CPU, CUDA: all_dims_out
+    MPS: all_dims_out_mps
```

Comment:
does mps implement all but not any?

Reply:
Oh sorry that's dead code. I looked briefly into adding it but the MPS code was fairly complex and I have no way to test so I decided against it. Removed in the latest commit.